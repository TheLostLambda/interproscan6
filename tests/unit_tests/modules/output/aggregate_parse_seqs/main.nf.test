nextflow_process {

    name "Test Process AGGREGATE_PARSED_SEQS"
    script "interproscan/modules/output/aggregate_parsed_seqs/main.nf"
    process "AGGREGATE_PARSED_SEQS"

    stage {
        copy "interproscan/scripts/output/aggregate_parsed_seqs.py"
        copy "tests/unit_tests/test_inputs/write_results/protein/parsed_seqs.json"
    }

    test("Should run successfully") {

        when {
            params {
            }
            process {
                """
                input[0] = ["../../../meta/tests/unit_tests/test_inputs/write_results/protein/parsed_seqs.json"]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
