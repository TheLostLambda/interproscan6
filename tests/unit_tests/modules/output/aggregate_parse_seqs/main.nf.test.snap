{
    "Should pass": {
        "content": [
            {
                "0": [
                    "parsed_seqs_aggregated.json:md5,6e5c21b388650baae39c1f55d18d6861"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-02T21:09:34.071541"
    },
    "Should run successfully": {
        "content": [
            {
                "0": [
                    "parsed_seqs_aggregated.json:md5,6e5c21b388650baae39c1f55d18d6861"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-11T13:48:52.965608"
    }
}